src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprSecCreateRecipe.java

```diff
@@ -132,35 +125,27 @@ public boolean init(Expression<?>[] exprs, int pattern, Kleenean isDelayed, Pars
 			Skript.error("Delays cannot be used within a 'create recipe' section.");
 			return false;
 		}
+		this.node = getParser().getNode();
+		rawExpr = result.expr;
 		return true;
 	}
 
 	@Override
 	protected Recipe @Nullable [] get(Event event) {
 		String name = providedName.getSingle(event);
 		if (name == null || name.isEmpty()) {
-			//Skript.error("The id for a recipe must not be null nor empty.");
+			error("The id for a recipe must not be null nor empty.");
 			return null;
 		}
```

Comment:
this shouldn't be an error, it should just return null. this matches the behavior of most expressions in skript.